### ISSUE TITLE
Matplotlib error when calling nipy.algorithms.registration.resample

### ISSUE THREAD

**Issue by peristeri** [nipy/nipy#430]
When calling `from nipy.algorithms.registration import resample` matplotlib is imported. It causes an exception when running on machines with no GUI. 

The culprit is `nipy.algorithms.registration.scripting` importing `matplotlib`. It might be related to #414 

```
File "/home/jcohen/sct/scripts/sct_resample.py", line 362, in main
resample()
File "/home/jcohen/sct/scripts/sct_resample.py", line 59, in resample
from nipy.algorithms.registration import resample
File "/home/jcohen/sct/python/lib/python2.7/site-packages/nipy/algorithms/registration/__init__.py", line 18, in <module>
from .scripting import space_time_realign, aff2euler
File "/home/jcohen/sct/python/lib/python2.7/site-packages/nipy/algorithms/registration/scripting.py", line 22, in <module>
import matplotlib.pyplot as plt
File "/home/jcohen/sct/python/lib/python2.7/site-packages/matplotlib/pyplot.py", line 114, in <module>
_backend_mod, new_figure_manager, draw_if_interactive, _show = pylab_setup()
File "/home/jcohen/sct/python/lib/python2.7/site-packages/matplotlib/backends/__init__.py", line 32, in pylab_setup
globals(),locals(),[backend_name],0)
File "/home/jcohen/sct/python/lib/python2.7/site-packages/matplotlib/backends/backend_qt4agg.py", line 18, in <module>
from .backend_qt5agg import FigureCanvasQTAggBase as _FigureCanvasQTAggBase
File "/home/jcohen/sct/python/lib/python2.7/site-packages/matplotlib/backends/backend_qt5agg.py", line 15, in <module>
from .backend_qt5 import QtCore
File "/home/jcohen/sct/python/lib/python2.7/site-packages/matplotlib/backends/backend_qt5.py", line 31, in <module>
from .qt_compat import QtCore, QtGui, QtWidgets, _getSaveFileName, __version__
File "/home/jcohen/sct/python/lib/python2.7/site-packages/matplotlib/backends/qt_compat.py", line 124, in <module>
from PyQt4 import QtCore, QtGui
ImportError: libSM.so.6: cannot open shared object file: No such file or directory
```

**Comment by matthew-brett** [comment]
Which version of nipy do you have:

```
>>> import nipy; print(nipy.__version__)
```

?

**Comment by peristeri** [comment]
The version is 0.4,0 on python 2.7 (miniconda)

**Comment by matthew-brett** [comment]
Can you upgrade? - I believe it is fixed with 0.4.1.  I don't know conda, but `pip install -U nipy` might work.

**Comment by peristeri** [comment]
I have not seen the error reappear. I guess upgrading solved the problem.

Thanks.
